commit eeef8b99e31d3b932ce2416f23820787f99004f0
Author: Devin Zuczek <devin.zuczek@gocadmium.com>
Date:   Fri Apr 8 11:24:25 2022 -0400

    CORE-4364: bitwise must be casted to bool

diff --git a/override_node_options.module b/override_node_options.module
index 7acfacf7d..8e4a12e26 100644
--- a/override_node_options.module
+++ b/override_node_options.module
@@ -238,7 +238,7 @@ function override_node_options_form_alter(&$form, &$form_state, $form_id) {
     $form['author']['name']['#access'] = user_access('override ' . $node->type . ' authored by option') || user_access('override all authored by option');
     $form['author']['date']['#access'] = user_access('override ' . $node->type . ' authored on option') || user_access('override all authored on option');
     if (array_key_exists('#access', $form['author'])) {
-      $form['author']['#access'] |= element_get_visible_children($form['author']);
+      $form['author']['#access'] |= (bool) element_get_visible_children($form['author']);
     }
     else {
       $form['author']['#access'] = element_get_visible_children($form['author']);
@@ -260,7 +260,7 @@ function override_node_options_form_alter(&$form, &$form_state, $form_id) {
     $form['options']['promote']['#access'] = user_access('override ' . $node->type . ' promote to front page option') || user_access('override all promote to front page option');
     $form['options']['sticky']['#access'] = user_access('override ' . $node->type . ' sticky option') || user_access('override all sticky option');
     if (array_key_exists('#access', $form['options'])) {
-      $form['options']['#access'] |= element_get_visible_children($form['options']);
+      $form['options']['#access'] |= (bool) element_get_visible_children($form['options']);
     }
     else {
       $form['options']['#access'] = element_get_visible_children($form['options']);
